#!/usr/bin/env python
PACKAGE = "opt_utils"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("rot_x",    double_t,    0, "Rotation wrt world x-axis", 0,  -1.570796, 1.570796)
gen.add("rot_y",    double_t,    0, "Rotation wrt world y-axis", 0,  -1.570796, 1.570796)
gen.add("rot_z",    double_t,    0, "Rotation wrt world z-axis", 0,  -1.570796, 1.570796)
gen.add("trs_x",    double_t,    0, "Translation wrt world x-axis", 0.0,  -5.0, 5.0)
gen.add("trs_y",    double_t,    0, "Translation wrt world y-axis", 0.0,  -5.0, 5.0)
gen.add("trs_z",    double_t,    0, "Translation wrt world z-axis", 0.0,  -5.0, 5.0)
gen.add("voxel_x",    double_t,    0, "Voxel x", 0.02,  0.0, 1.0)
gen.add("voxel_y",    double_t,    0, "Voxel y", 0.02,  0.0, 1.0)
gen.add("voxel_z",    double_t,    0, "Voxel z", 0.02,  0.0, 1.0)
#size_enum = gen.enum([ gen.const("Save",      int_t, 0, "No operation", ),
#                       gen.const("DoNotSaveYet",        int_t, 1, "To save the matrix transf file", )],
#                     "select Save if the clouds are aligned to save the matrix!")
#gen.add("stop", int_t, 0, "Save", 0, 0, 1, edit_method=size_enum)

exit(gen.generate(PACKAGE, "opt_utils", "ManualCalibration"))
